Tree: Make fixes to detach ID overrides

alex-pardes opened the pull request:
## Description

This fixes a bug where lineage was not added for detaches with a defined `detachIdOverride`.

alex-pardes committed:
Add failing tests

alex-pardes committed:
Merge remote-tracking branch 'fluid/main' into detach-id-override

alex-pardes committed:
Merge remote-tracking branch 'fluid/main' into detach-id-override

alex-pardes committed:
Merge remote-tracking branch 'fluid/main' into detach-id-override

yann-achard-MS committed:
test(tree2): add tests for empty cell ordering during compose (#18738)

CraigMacomber committed:
Tag asserts (#18743)

## Description

Tag asserts for release

yann-achard-MS committed:
fix(tree2): bug in transfer of nested changes (#18737)

anthony-murphy committed:
Mark Server Protocol Handler Alpha (#18747)

This is used on the alpha surface of client, so also need to be marked
alpha:

https://github.com/microsoft/FluidFramework/blob/b3e6b6bbd57222548a5d914da1220cb2c1a89be0/packages/loader/container-loader/src/protocol.ts#L9

Co-authored-by: Tony Murphy <anthonm@microsoft.com>

kian-thompson committed:
Add some comments around protection of resubmitting same op (#18708)

We have mechanisms in place that help prevent ops from being resubmitted
if they were successfully sequenced by service (but our client didn't
have time to see the ack before disconnecting).
Added comments in some key places so people interested can quickly find
the code that prevents this.

Example scenario:

| **Step** | **clientId value** | **PendingStateManager** |

|--------------------------------------------|----------------|---------------------|
| Client A sends op 1 | Client A | [op 1] |
| Client A disconnects | Client A | [op 1] |
| Client A reconnects as Client B | Client A | [op 1] |
| **Start wait to see "leave" op from Client A** | Client A | [op 1] |
| See ack for op 1 | Client A | **[]** |
| See "leave" for Client A | **Client B** | [] |

agarwal-navin committed:
GC: Ignore incoming changes to data store that has been deleted (#18731)

After a data store is deleted as a result of sweep, further modification
to it must be prevented to cause any data loss. This has been added via
a GC sweep op which will be received by all clients
https://github.com/microsoft/FluidFramework/pull/18632.
However, due to an unexpected bug, a client could have a (to be) deleted
data store loaded and before it receives the GC sweep op, this data
store can send an op or signal. These ops must be ignored after the data
store is deleted to not cause document corruption. This PR adds this
logic. An error is still logged and it can be used to identify if such
case ever happens.

AB#6113

anthony-murphy committed:
Beta bump server protocol defs (#18750)

bump the version of protocol definitions in server

---------

Co-authored-by: Tony Murphy <anthonm@microsoft.com>

yann-achard-MS committed:
test(tree2): enable passing fuzz seed (#18745)

yann-achard-MS committed:
fix(tree2): apply move effects to MovePlaceholder marks (#18753)

jzaffiro committed:
Refactor of IntervalCollection.change and IntervalCollection.changeProperties  (#18676)

Refactors the change method to include the functionality of changing the
endpoints as well as the properties of an interval. It also updates the
signature of the (new) change method to align with the changes made to
the add signature in #17165. The changeProperties method and the old
signature of change will be deprecated and eventually removed.

AB#5738

---------

Co-authored-by: Abram Sanderson <Abram.sanderson@gmail.com>

taylorsw04 committed:
Move IdCompressor into a separate package (#18749)

## Description

This PR moves the ID compressor into a separate package, cleaning up the
dependency structure and making it possible to depend on it in places
where depending on the runtime is disallowed or inappropriate.

alex-pardes committed:
Fix lineage for detachIdOverride

CraigMacomber committed:
tree2: Add enum example (#18732)

## Description

Add example of unions like enums.

Future work could polish up this utility and we could export it. For now
this shows how it could be done.

Josmithr committed:
docs(id-compressor): Update README to be formatted as a package README (#18756)

Updates the existing documentation (which was previously an interior
directory overview doc) to be a package README.
- Updates existing formatting to better match our standard patterns
- Adds basic Fluid Framework package instructions using our
`markdown-magic` pragmas

alex-pardes committed:
Cleanup

yann-achard-MS committed:
test(tree2): Enable sequence state-based axiom tests with a limited config (#18741)

yann-achard-MS commented on experimental/dds/tree2/src/feature-libraries/sequence-field/rebase.ts:
```diff
@@ -154,11 +154,7 @@ function rebaseMarkList<TNodeChange>(
 
 		// Inverse attaches do not contribute to lineage as they are effectively reinstating
 		// an older detach which cells should already have any necessary lineage for.
```
Should this comment be removed?

markfields committed:
GC: Fix bug in new Two-Stage Sweep feature (#18734)

Fixes a bug in #18559 where SweepReady nodes would not be Tombstoned
when Sweep was disabled.

markfields committed:
GC: Simplify configuration for when to enable sweep and for which documents (#18725)

Fixes AB#6481.

In light of #18559, Tombstone and Sweep should always be enabled in
lockstep in terms of which documents are eligible. Today this is
determined by two independent options `gcTombstoneGeneration` and
`gcSweepGeneration`. We are consolidating on a single `gcGeneration`
(with a back-compat measure for old docs with only
`gcTombstoneGeneration`).

The semantics follow the same concept: Any document stamped with the
matching generation value that's current passed will be eligible for
both the Tombstone and Delete stages of Sweep Phase. Importantly, if no
value is passed in, Sweep is considered enabled for that document.

Then in order to configure whether to run the Delete stage in a given
session, there's a new GC option called `enableGCSweep` which can be set
to `true` (default is don't delete, only Tombstone).

### Additional behavior change:

* Small bug fix: Tombstone Enforcement was always false in the first
session for a container if a tombstoneGeneration option was passed

alex-pardes committed:
Remove stale comment

dannimad committed:
release notes for minor 7.4 (#18740)

As part of preparation for next 7.4.0 minor release.

---------

Co-authored-by: Tyler Butler <tyler@tylerbutler.com>

tylerbutler committed:
build(protocol-definitions): Produce correct types in ESM build (#18771)

Updates the protocol-definitions build to produce both CJS and ESM with
correct types. Also adds are-the-types-wrong to verify the types.

znewton committed:
Gitrest: whole summary unit/regression tests (#18760)

## Description

Adds a suite of unit tests to the Gitrest GitWholeSummaryManager.

## Reviewer Guidance

- I generated the `examples.ts` data from an open source app that I
authored myself. The content is Lorem Ipsum based, so there is no
"customer content" here.

sonalideshpandemsft committed:
(odsp-client): Mark e2e test package as non-private  (#18786)

This PR marks the odsp-client e2e test package as non-private

tylerbutler committed:
build: Remove declarationDir setting from tsconfigs (#18792)

Projects no longer need to explicitly define a declarationDir in their
tsconfig, and those that did were causing ESM type output to be placed
alongside CJS source in ./dist. This PR removes the declarationDir
setting from all tsconfigs.

dannimad committed:
Tag asserts (#18790)

Remaining asserts tagging  for 7.4 minor release

dannimad committed:
Generate change logs for minor release 7.4 (#18787)

Following
https://eng.ms/docs/experiences-devices/opg/office-shared/fluid-framework/fluid-framework-internal/fluid-framework/docs/on-call/release/release-minor

Running  `pnpm flub generate changelog -g client`

yann-achard-MS committed:
Feedback and more

yann-achard-MS committed:
Merge branch 'detach-id-override' of https://github.com/alex-pardes/fluidframework into pr/alex-pardes/18742

yann-achard-MS committed:
Merge branch 'main' into pr/alex-pardes/18742

yann-achard-MS committed:
Fix and docs

yann-achard-MS committed:
Dedup error codes
